File: Backend/src/Agro_Trade.Domain/Entities/Valoracion.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace Agro_Trade.Domain.Entities
{
    public class Valoracion
    {
        [Key]
        public int IdValoracion { get; set; }
        public int IdPedido { get; set; }
        public Pedido Pedido { get; set; } = null!;

        public int IdUsuarioCliente { get; set; }
        public Usuario UsuarioCliente { get; set; } = null!;

        public int IdProveedor { get; set; }
        public Proveedor Proveedor { get; set; } = null!;

        public string? TipoValoracion { get; set; }
        public int Puntuacion { get; set; } // 1-5
        public string? Comentario { get; set; }
        public DateTime FechaValoracion { get; set; } = DateTime.UtcNow;
    }
}

File: Frontend/agrotrade_frontend/lib/models/Consumidor/interaccion/valoracion.dart
class ValoracionConsumidor {
  final int idValoracion;
  final int idPedido;
  final int idUsuarioCliente;
  final int idProveedor;
  final String? tipoValoracion;
  final int puntuacion;
  final String? comentario;
  final DateTime fechaValoracion;

  const ValoracionConsumidor({
    required this.idValoracion,
    required this.idPedido,
    required this.idUsuarioCliente,
    required this.idProveedor,
    this.tipoValoracion,
    required this.puntuacion,
    this.comentario,
    required this.fechaValoracion,
  });

  factory ValoracionConsumidor.fromJson(Map<String, dynamic> json) {
    return ValoracionConsumidor(
      idValoracion: json['idValoracion'] as int,
      idPedido: json['idPedido'] as int,
      idUsuarioCliente: json['idUsuarioCliente'] as int,
      idProveedor: json['idProveedor'] as int,
      tipoValoracion: json['tipoValoracion'] as String?,
      puntuacion: json['puntuacion'] as int? ?? 5,
      comentario: json['comentario'] as String?,
      fechaValoracion: json['fechaValoracion'] != null
          ? DateTime.tryParse(json['fechaValoracion'] as String) ?? DateTime.now()
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idValoracion': idValoracion,
      'idPedido': idPedido,
      'idUsuarioCliente': idUsuarioCliente,
      'idProveedor': idProveedor,
      'tipoValoracion': tipoValoracion,
      'puntuacion': puntuacion,
      'comentario': comentario,
      'fechaValoracion': fechaValoracion.toIso8601String(),
    };
  }
}
